test_mavros/include/tests/offboard_control.h

```diff
@@ -0,0 +1,470 @@
+/**
+ * @brief Offboard control test
+ * @file offboard_control.h
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ * @author Andre Nguyen <andre-phu-van.nguyen@polymtl.ca>
+ *
+ * @addtogroup sitl_test
+ * @{
+ */
+/*
+ * Copyright 2015 Nuno Marques, Andre Nguyen.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros.h>
+#include <sitl_test/sitl_test.h>
+#include <sitl_test/test_type.h>
+
+#include <geometry_msgs/Point.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+
+namespace testtype {
+/**
+ * @brief Offboard controller tester
+ *
+ * Tests offboard position, velocity and acceleration control
+ *
+ */
+class OffboardControl {
+public:
+	OffboardControl() :
+		nh_sp("~"),
+		mode("position"),
+		shape("square")
+	{ };
+
+	/* -*- main routine -*- */
+
+	void spin(int argc, char *argv[]) {
+		/**
+		 * @brief Setpoint control mode selector
+		 *
+		 * Available modes:
+		 * - position
+		 * - velocity
+		 * - acceleration
+		 */
+		nh_sp.param<std::string>("mode", mode, "position");
+
+		/**
+		 * @brief Setpoint path shape selector
+		 *
+		 * Available shapes:
+		 * - square
+		 * - circle
+		 * - eight
+		 * - ellipse (in 3D space)
+		 */
+		nh_sp.param<std::string>("shape", shape, "square");
+
+		ros::Rate loop_rate(10);
+
+		ROS_INFO("SITL Test: Offboard control test running!");
+
+		if (mode.compare("position") == 0) {
```

Comment:
You can do `mode == "position"`. Also there nothing to do if mode incorrect,
so do first bunch of checks about mode, then shape.